tools/react-integration-tester/src/shared.ts

```diff
@@ -0,0 +1,255 @@
+import { spawn } from 'node:child_process';
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'node:fs';
+import { dirname, join } from 'node:path';
+import { type PackageJson } from 'nx/src/utils/package-json';
```

Comment:
this is internal only -> no need to install `nx` for 3rd party, will extract these types later to have "clean" dep tree